### ISSUE TITLE
Why does GCM needs to connect to mtalk.google.com ?

### ISSUE THREAD

**Issue by theofruitrouge** [microg/android_packages_apps_GmsCore#569]
Following issue #566, I am wondering if microG creates a local redirection to intercept traffic intended to mtalk.google.com or is it connecting to that domain ? If so, why ?

**Comment by ArchangeGabriel** [comment]
Because GCM depends on Google servers? If you don’t know how GCM works, no problem, I’ll explain it. ;)

**Comment by theofruitrouge** [comment]
@ArchangeGabriel Indeed I don't know how GCM works… I don't mean to waste your time but if you want you can explain me !

**Comment by ArchangeGabriel** [comment]
No, FCM builds upon GCM, it works the same way w.r.t. my above explanation. As far as proprietary apps are concerned, there is little hope to change anything here, but for FLOSS software maybe a self-hosted replacement could be done.

**Comment by flatsiedatsie** [comment]
I was suprised to see a connection to `mtalk.google.com` after getting /e/ os.

The /e/ os overview page starts with this header:

```
/e/OS is a complete, fully “deGoogled”, mobile ecosystem
```

But... it's not?

While there is surely a technical reason, as mentioned above, it smells a little bit like false advertising to me.

**Comment by ArchangeGabriel** [comment]
As a (now deleted) comment revived this, I take the opportunity to mention that https://unifiedpush.org/ is the long awaited solution for FLOSS software.
